File: largest-unique-number/largest-unique-number.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        count = Counter(nums)
        ans = -math.inf
        for key, val in count.items():
            if val == 1:
                ans = max(ans, key)
        
        return (-1 if ans == -math.inf else ans)

File: largest-unique-number/largest-unique-number.java
class Solution {
    public int largestUniqueNumber(int[] nums) {
        int largest = Integer.MIN_VALUE;
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        for(int key: map.keySet()){
            int freq = map.get(key);
            if(freq == 1){
                largest = Math.max(largest, key);
            }
        }
        
        return largest == Integer.MIN_VALUE ? -1 : largest;
    }
}